src/stable/test/deep_equal.ts

```diff
@@ -0,0 +1,68 @@
+import { jsonStringify } from '#lib/json';
+
+/**
+ * @internal
+ *
+ * Checks if two objects are deeply equal.
+ *
+ * @param obj1 The first object to compare.
+ * @param obj2 The second object to compare.
+ * @returns `true` if the objects are deeply equal, `false` otherwise.
+ *
+ * @remarks
+ * It first serializes both objects using our custom jsonStringify function to handle potential problems with Principal
```

Comment:
Comments should explain that not only Principal is the problem

Reply:
done